### ISSUE TITLE
Replace `LexException: f-string: trailing junk in field` with a more descriptive error message

### ISSUE THREAD

**Issue by shadowrylander** [hylang/hy#2310]
Hello!

Recently, I came across an issue where the following gave me the error in the title:

```hy
f"{(fun).attr}"
```

@Kodiologist notified me [here](https://stackoverflow.com/questions/72627566/lexexception-f-string-trailing-junk-in-field?noredirect=1#comment128331107_72627566) that `(fun).attr` is not valid `hy` syntax, but if I recall correctly, outside an `f-string`, this error message would've been stated explicitly.

Basically, what I'm asking is, would it be possible to replace `LexException: f-string: trailing junk in field` with a more explicit error message, or the exact error message, as a regular syntax error outside an `f-string` would give?

Thank you kindly for your consideration on the matter!

**Comment by Kodiologist** [comment]
> or the exact error message, as a regular syntax error outside an f-string would give

The parser never gets as far as "cannot access attribute on anything other than a name" because `.nodes.nixos-22-05.locked.rev` doesn't get parsed. Only one form is allowed in a replacement field.

**Comment by Kodiologist** [comment]
So, I think the current error message is probably the right one.

**Comment by shadowrylander** [comment]
Huh; can it at least tell the user where the error is, or is it still because `.attr` doesn't get parsed?

**Comment by Kodiologist** [comment]
It looks like error is already located correctly, after the form that was successfully parsed:

```
    f"{(fun).attr}"
            ^
hy.reader.exceptions.LexException: f-string: trailing junk in field
```

**Comment by shadowrylander** [comment]
Interesting; for me, it's just pointing at the `f`, so:
```hy
    f"{(fun).attr}"
    ^
hy.reader.exceptions.LexException: f-string: trailing junk in field
```
I'm using the [`rich` module's traceback function](https://rich.readthedocs.io/en/stable/reference/traceback.html#rich.traceback.install), so that may be the issue here. I'll go ahead and close this one, then; sorry for the noise!

**Comment by Kodiologist** [comment]
That's curious. It could be a Hy version difference (although I think you're pretty close to master right now), a bug in Hy's traceback construction, or a bug in `rich`.

**Comment by shadowrylander** [comment]
I'm actually using version `1.0a4`, the same as in the [`nixpkgs`](https://github.com/nixos/nixpkgs/blob/nixos-22.05/pkgs/development/python-modules/hy/default.nix) repo. How can we determine where it's coming from?

**Comment by Kodiologist** [comment]
Oh, it is a Hy version difference, then. The arrow is indeed misplaced in 1.0a4.

**Comment by shadowrylander** [comment]
Got it; I'm assuming that `1.0a5` (I think?) should fix this then, right?

**Comment by Kodiologist** [comment]
It's fixed in master.

**Comment by shadowrylander** [comment]
Yes, but I'd prefer a version release, as there seem to be significant syntax changes since `1.0a4` and the current `master`.

**Comment by Kodiologist** [comment]
You're in luck: #2298.

**Comment by shadowrylander** [comment]
Ooh! I'll be waiting eagerly! 😸
